scenarios/kubernetes_e2e.py

```diff
@@ -44,40 +44,54 @@ def check(*cmd):
     subprocess.check_call(cmd)
 
 
-def sig_handler(_signo, _frame):
-    """Stops container upon receive signal.SIGTERM and signal.SIGINT."""
-    print >>sys.stderr, 'signo = %s, frame = %s' % (_signo, _frame)
-    check('docker', 'stop', CONTAINER)
+def setup_signal_handlers(container):
+    """Establish a signal handler to kill 'container'."""
+    def sig_handler(_signo, _frame):
+        """Stops container upon receive signal.SIGTERM and signal.SIGINT."""
+        print >>sys.stderr, 'signo = %s, frame = %s' % (_signo, _frame)
+        check('docker', 'stop', container)
+
+    signal.signal(signal.SIGTERM, sig_handler)
+    signal.signal(signal.SIGINT, sig_handler)
 
 
 def kubekins(tag):
     """Return full path to kubekins-e2e:tag."""
     return 'gcr.io/k8s-testimages/kubekins-e2e:%s' % tag
 
-
-def main(args):
-    """Set up env, start kubekins-e2e, handle termination. """
-    # pylint: disable=too-many-locals
-
-    # dockerized-e2e-runner goodies setup
-    workspace = os.environ.get('WORKSPACE', os.getcwd())
-    artifacts = '%s/_artifacts' % workspace
-    if not os.path.isdir(artifacts):
-        os.makedirs(artifacts)
-
-    try:  # Pull a newer version if one exists
-        check('docker', 'pull', kubekins(args.tag))
+def pull_kubekins(tag):
+    """Try to pull a newer version of kubekins image."""
+    try:
+        check('docker', 'pull', kubekins(tag))
     except subprocess.CalledProcessError:
         pass
 
-    print 'Starting %s...' % CONTAINER
-
-    cmd = [
+def append_env(key, val, docker):
+    """Map env to docker or to environ."""
+    if docker:
```

Comment:
https://github.com/kubernetes/test-infra/pull/2001 passes on my workstation with both --mode=docker and --mode=local

There are some gross hacks for the --mode=local that will be better once I get in https://github.com/kubernetes/test-infra/pull/1657#issuecomment-282083789